File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.py
    # 123   1234
    # 223   2234
    # 333   3334
    #       4444
def solution(n, left, right):
    answer = []
    for i in range(left,right+1):
        div,mod = divmod(i,n)
        answer.append(max(div,mod)+1)
    return answer

File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, long long left, long long right) {
    vector<int> answer;
    long long div = 0,  mod = 0;
    for (long long i = left; i <= right; i++){
        div = i / n;
        mod = i % n;
        if (div < mod) {
            answer.push_back(mod+1);
        }
        else {
            answer.push_back(div+1);
        }
    }
    return answer;
}